### PR TITLE
crypto-square: tests are now generated

### PR DESCRIPTION
**Pull request by ajwann**
## Why?
All tests should be generated as per #396

## What?
This PR contains a generator for the crypto-square exercise.

## How Has This Been Tested?
```bin/generate crypto-square```
```rake test:crypto-square```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or enhancement that would cause existing functionality to change)
- [ ] Gardening (code and/or documentation cleanup)

## References and Closures
#396

## Checklist:

- [x] I have read the **CONTRIBUTING** document.
- [ ] My change relies on a pending issue/pull request
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: exercises/crypto-square/.meta/generator/crypto_square_case.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+require 'generator/exercise_case'
+
+class CryptoSquareCase < Generator::ExerciseCase
+
+  # TODO: remove guard clause when #encoded method
+  # is added to crypto_square.rb
```
Comment (ajwann) [inline exercises/crypto-square/.meta/generator/crypto_square_case.rb]: So the canonical data is a bit off for this exercise. There are several tests in the canonical data that are expecting a method named ```encode``` or possibly ```encoded``` to be defined on the ```Crypto``` class. However, this method is missing from ```crypto_square.rb```

File: exercises/crypto-square/.meta/generator/crypto_square_case.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  def ciphertext_method_call
+    if canonical.expected.include?(' ')
+      'crypto.normalize_ciphertext'
+    else
+      'crypto.ciphertext'
+    end
+  end
+
+  # TODO: remove when canonical data is fixed
```
Comment (ajwann) [inline exercises/crypto-square/.meta/generator/crypto_square_case.rb]: The canonical data for one test also has an extra space, which causes the assertion to fail.

File: exercises/crypto-square/.meta/solutions/crypto_square.rb [hunk 3]
```diff
@@ -9,6 +13,7 @@ def normalize_plaintext
   end
 
   def plaintext_segments
+    return [] if normalize_plaintext == ''
```
Comment (ajwann) [inline exercises/crypto-square/.meta/solutions/crypto_square.rb]: The canonical data had a test which asserts that an empty array passed into ```Crypto.plaintext_segments``` will return an empty array. However, the current ```Crypto``` class just blows up when it tries to call ```each_slice``` with a size of 0.

File: exercises/crypto-square/.meta/solutions/crypto_square.rb [hunk 4]
```diff
@@ -20,6 +25,7 @@ def size
   end
 
   def ciphertext
+    return '' if normalize_plaintext == ''
```
Comment (ajwann) [inline exercises/crypto-square/.meta/solutions/crypto_square.rb]: The canonical data also had a test which asserts that an empty array passed into ```Crypto.ciphertext``` will return an empty string. However, the current ```Crypto``` class still just blows up when it tries to call ```each_slice``` with a size of 0.

File: exercises/crypto-square/crypto_square_test.rb [hunk 5]
```diff
@@ -1,105 +1,97 @@
...
+    assert_equal [], crypto.plaintext_segments
   end
 
-  def test_size_of_slightly_larger_square
+  def test_4_character_plaintext_results_in_an_2x2_rectangle
     skip
-    crypto = Crypto.new('123456789')
-    assert_equal 3, crypto.size
+    crypto = Crypto.new('Ab Cd')
+    assert_equal ["ab", "cd"], crypto.plaintext_segments
```
Comment (ajwann) [inline exercises/crypto-square/crypto_square_test.rb]: So these tests seemed to have assertion that differed from what the canonical data wanted.
I went with the canonical data's choice of assertions, because I figured it was more up-to-date.

**Comment by nholden** [comment]
Hi, @ajwann! It looks like this PR is using an older version of the canonical data. I submitted https://github.com/exercism/ruby/pull/738, which uses the latest version of the canonical data. Would you mind taking a look and letting me know what you think?

**Comment by Insti** [comment]
It looks like this is a case of "great minds think alike" and we have 2 PRs to generate tests for the same problem.

#738 Looks like a currently more complete and functional solution so I'll close this one.

Thanks @ajwann for your work on this. :heart:

**Comment by ajwann** [comment]
@nholden I should have though to pull from master in my x-common repo! Oh well, nice work with the PR.
